src/g2.rs

```diff
@@ -26,6 +30,27 @@ pub struct G2Affine {
     infinity: Choice,
 }
 
+#[cfg(feature = "canon")]
+impl<S: Store> Canon<S> for G2Affine {
+    fn write(&self, sink: &mut impl Sink<S>) -> Result<(), S::Error> {
+        self.x.write(sink)?;
+        self.y.write(sink)?;
```

Comment:
Why not compressing the point as bytes? That will reduce the size and grant us a check when we use read.

Anyway, LGTM.